authorization/role_assignment_service_blackbox_test.go

```diff
@@ -0,0 +1,240 @@
+
+import (
+	"testing"
+
+	"github.com/fabric8-services/fabric8-auth/authorization"
+	"github.com/fabric8-services/fabric8-auth/authorization/models"
+	"github.com/fabric8-services/fabric8-auth/authorization/role"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/gormtestsupport"
+	testsupport "github.com/fabric8-services/fabric8-auth/test"
+
+	"github.com/goadesign/goa/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type roleAssignmentServiceBlackboxTest struct {
+	gormtestsupport.DBTestSuite
+	assignmentService authorization.RoleAssignmentService
+}
+
+func TestRunRoleAssignmentServiceBlackboxTest(t *testing.T) {
+	suite.Run(t, &roleAssignmentServiceBlackboxTest{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *roleAssignmentServiceBlackboxTest) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	modelService := models.NewRoleAssignmentModelService(s.DB, s.Application)
+	s.assignmentService = authorization.NewRoleAssignmentService(modelService, s.Application)
+}
+func (s *roleAssignmentServiceBlackboxTest) TestGetIdentityRoleByResource() {
+	t := s.T()
+	identityRole, err := testsupport.CreateRandomIdentityRole(s.Ctx, s.DB)
+	require.NoError(t, err)
+	require.NotNil(t, identityRole)
+
+	identityRoles, err := s.assignmentService.ListByResource(s.Ctx, identityRole.Resource.ResourceID)
+	require.NoError(t, err)
+	require.Equal(t, true, len(identityRoles) == 1)
+	require.Equal(t, identityRole.Resource.ResourceID, identityRoles[0].Resource.ResourceID)
+}
+
+func (s *roleAssignmentServiceBlackboxTest) TestGetMultipleIdentityRoleByResourceInherited() {
+
+	/*
+		Create resource "AreaDev"
+		Create child resource "AreaAuth"
+
+		Create 1 assigned role for "AreaDev"
+		Create 2 assigned roles for "AreaAuth"
+
+		List all roles for "AreaAuth" : Should return the above 3 assigned roles.
+	*/
+
+	t := s.T()
+	resourceOwner := testsupport.TestIdentity2
+	err := testsupport.CreateTestIdentityAndUserInDB(s.DB, &resourceOwner)
+	require.NoError(s.T(), err)
+
+	areaResourceType, err := s.Application.ResourceTypeRepository().Lookup(s.Ctx, "openshift.io/resource/area")
+	require.NoError(s.T(), err)
+
+	parentResourceRef, err := testsupport.CreateTestResource(s.Ctx, s.DB, *areaResourceType, "AreaDev")
+	require.NoError(s.T(), err)
+
+	resourceRef, err := testsupport.CreateInheritedTestResource(s.Ctx, s.DB, *areaResourceType, "AreaAuth", *parentResourceRef)
+	require.NoError(s.T(), err)
+
+	roleRef, err := testsupport.CreateTestRole(s.Ctx, s.DB, *areaResourceType, "collab")
+	require.NoError(s.T(), err)
+
+	var createdIdentityRoles []role.IdentityRole
+
+	// creating an AssignedRole for a parent resource
+	identityRoleRefUnrelated, err := testsupport.CreateTestIdentityRole(s.Ctx, s.DB, *parentResourceRef, *roleRef)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), identityRoleRefUnrelated)
+
+	// creating an AssignedRole for an inherited resource
```

Comment:
I suggest you refer to this as child resource, not inherited resource as _inherited_ is not really correct in this context (this applies to all similar references throughout this PR).

Reply:
7b752e5